src/lib/optimizer/strategy/column_pruning_rule.cpp

```diff
@@ -242,6 +242,36 @@ void recursively_gather_required_expressions(
   }
 }
 
+void annotate_join_unused_inputs(
+    const std::shared_ptr<AbstractLQPNode>& node,
+    const std::unordered_map<std::shared_ptr<AbstractLQPNode>, ExpressionUnorderedSet>& required_expressions_by_node) {
+  // Identify whether the left or right side of the given join node are not needed anywhere further up in the LQP
+  // If one of the sides is not used, add this information to the join node
+  auto join_node = std::dynamic_pointer_cast<JoinNode>(node);
+
+  auto left_input_is_used = false;
+  auto right_input_is_used = false;
+  for (const auto& output : node->outputs()) {
+    for (const auto& required_expression : required_expressions_by_node.at(output)) {
+      if (expression_evaluable_on_lqp(required_expression, *node->left_input()))
+        left_input_is_used = true;
+      if (expression_evaluable_on_lqp(required_expression, *node->right_input()))
+        right_input_is_used = true;
+    }
+  }
+  DebugAssert(left_input_is_used || right_input_is_used, "Did not expect a useless join");
```

Comment:
Can this yield into problems when running a release build?

Reply:
Could you provide a more detailed description of the failure cause in the error message?